### DISCUSSION TITLE
How to use search and multi language mode?

### DISCUSSION THREAD

**Discussion by railduino** [datenstrom/community#315] [Ask a question] [answered]
My website is running fine with .../en/1-... and .../de/1-... plus /default/
After installing the search plugin I can call .../de/search/query:foo/ and it works.
But when I fill in the search form on the content page, the form wants to POST to /search/ instead of /de/search/
I believe the generating line is in navigation-search.php in the line "<?php echo $yellow->page->base.$yellow->config->get("searchLocation") ?>"
Is there a way to add the language prefix? Currently I always get a 404 because it tries to call /search/.
BTW: Thank you very much -- excellent work!

**Comment by markseuffert** [comment, marked as answer]
Yes you're right. 

The API has a function to get the prefix. You can change `page->base` to `page->getBase(true)` where needed. The argument tells the function to return the multi language base, for example `/en` or `/de`. Hope this helps.

**Comment by railduino** [comment]
Thank you - works as expected.
